### PR TITLE
Make contentlisting work with plone.app.contenttypes collections

### PR DESCRIPTION
**Pull request by kiorky**
cc @ebrehault @vangheem

### CODE REVIEW

**Comment by kiorky** [comment]
This is to make folder_contents works both on plone.app.contenttypes & folders on plone4

**Comment by thet** [comment]
nice! for plone.app.contenttypes, I'm working on something similar : https://github.com/plone/plone.app.contenttypes/pull/202

**Comment by kiorky** [comment]
yep but that's for plone5, my little commit makes the listing beginning to work with the branch for plone4 :)

**Comment by kiorky** [comment]
i think your approach in the long term is better, it is just way too invasive rather than mine in the idea that i dont had the time to do the nice refactor you are going on, so i did that proxy trick to make things assemble on plone4.3

**Comment by thet** [comment]
oh, sorry, i have mistaken folder_contents as folder_listing, which is just not true. so your fix makes perfect sense.

recently i was wondering, if we could get rid of plone.app.content in Plone 5 at all.

File: plone/app/content/browser/foldercontents.py [hunk 1]
```diff
@@ -19,6 +22,50 @@
...
+        ignored = ['modified', 'portal_type', 'getURL',
+                   'is_folderish', 'getObjSize']
+        ret = getattr(self.obj, attr)
+        call = self.cl and (attr not in ignored)
+        if call and not isinstance(ret, (basestring, int, bool,
+                                         DateTime.DateTime,
+                                         datetime.datetime,
+                                         datetime.time,
+                                         float, list, set, tuple)):
+            ret = ret()
```
Comment (thet) [inline plone/app/content/browser/foldercontents.py]: is this clause the same like doing it like so?
```
from Products.CMFPlone.utils import safe_callable
if safe_callable(ret):
    ret = ret()
```

```

File: plone/app/content/browser/foldercontents.py [hunk 2]
```diff
@@ -19,6 +22,50 @@
 from plone.app.content.browser.tableview import Table, TableBrowserView
 
 
+try:
+    from plone.app.contentlisting.interfaces import IContentListingObject
+    HAS_CL = True
+except:
```
Comment (thet) [inline plone/app/content/browser/foldercontents.py]: ```except ImportError:```

**Comment by thet** [comment]
merge @ebrehault ? you have more insight...

File: plone/app/content/browser/foldercontents.py (hunk 1 above)
Comment (kiorky) [inline plone/app/content/browser/foldercontents.py]: maybe yes, i should read the code of safe_callable :)

File: plone/app/content/browser/foldercontents.py (hunk 2 above)
Comment (kiorky) [inline plone/app/content/browser/foldercontents.py]: fuck, how can i have missed that ! sorry, i ll push something

**Comment by kiorky** [comment]
For the use case, we use folder_contents to post to a custom view via portal_actions (the user select boxes on folder_contents) and that is posted to a custom view via a custom action in folder_buttons.

The view is here; https://github.com/RBINS/rbins_masschange/blob/master/src/rbins_masschange/browser/masschange.py

I know that there is a view on plone5 which may have all if not better features of this view, but this view is older and works on plone4 :p

File: plone/app/content/browser/foldercontents.py (hunk 1 above)
Comment (kiorky) [inline plone/app/content/browser/foldercontents.py]: addressed

File: plone/app/content/browser/foldercontents.py (hunk 2 above)
Comment (kiorky) [inline plone/app/content/browser/foldercontents.py]: addressed

**Comment by thet** [comment]
what i still don't understand is: is your collection type folderish? in  ``plone.app.content/plone/app/content/browser/foldercontents.zcml`` the ``folder_contents`` view is only bound to `` for="Products.CMFCore.interfaces._content.IFolderish``, which collections from plone.app.contenttypes aren't, IIRC.

**Comment by kiorky** [comment]
@thet you didnt see plone/plone.app.layout/pull/40 didnt you

**Comment by thet** [comment]
@jensens @pbauer @ebrehault  do someone of you have some time to look at this? i'm currently too busy to handle this.

**Comment by thet** [comment]
see discussion at: https://github.com/plone/plone.app.layout/pull/40

**Comment by jensens** [comment]
I close this one, if you think this is wrong please reopen and comment.

**Pull request closed without merging**